src/common/cron.cc

```diff
@@ -57,10 +62,21 @@ bool Cron::IsTimeMatch(const tm *tm) {
       tm->tm_mon == last_tm_.tm_mon && tm->tm_wday == last_tm_.tm_wday) {
     return false;
   }
+
+  auto match = [](int current, int val, bool interval) {
+    if (val == -1) return true;
+    if (interval) return current % val == 0;
+    return val == current;
+  };
+
   for (const auto &st : schedulers_) {
-    if ((st.minute == -1 || tm->tm_min == st.minute) && (st.hour == -1 || tm->tm_hour == st.hour) &&
-        (st.mday == -1 || tm->tm_mday == st.mday) && (st.month == -1 || (tm->tm_mon + 1) == st.month) &&
-        (st.wday == -1 || tm->tm_wday == st.wday)) {
+    bool minute_match = match(tm->tm_min, st.minute, st.minute_interval);
+    bool hour_match = match(tm->tm_hour, st.hour, st.hour_interval);
+    bool mday_match = match(tm->tm_mday, st.mday, st.mday_interval);
+    bool month_match = match(tm->tm_mon - 1, st.month, st.month_interval);
```

Comment:
```suggestion
    bool month_match = match(tm->tm_mon + 1, st.month, st.month_interval);
```

Reply:
But If I do that, my tests for the month day will fail. So I guess my tests are bad then? According to crontab.guru, a crontab like `0 0 1 */4 *` will result in the following:

```
next  at 2024-09-01 00:00:00
then at 2025-01-01 00:00:00
then at 2025-05-01 00:00:00
then at 2025-09-01 00:00:00
then at 2026-01-01 00:00:00
```

https://crontab.guru/#0_0_1_*/4_*